src/voice-search/voice-search.ts

```diff
@@ -0,0 +1,162 @@
+import {
+  Component,
+  Input,
+  Inject,
+  forwardRef,
+  NgZone,
+  ContentChild,
+  ElementRef,
+  TemplateRef,
+  OnInit,
+} from '@angular/core';
+
+import { connectVoiceSearch } from 'instantsearch.js/es/connectors';
+import { VoiceListeningState } from 'instantsearch.js/es/lib/voiceSearchHelper';
+import { BaseWidget } from '../base-widget';
+import { NgAisInstantSearch } from '../instantsearch/instantsearch';
+import { noop } from '../utils';
+
+type TemplateContext = {
+  status: string;
+  transcript: string;
+  isSpeechFinal: boolean;
+  errorCode?: string;
+  isBrowserSupported: boolean;
+  isListening: boolean;
+};
+
+type State = {
+  isBrowserSupported: boolean;
+  isListening: boolean;
+  toggleListening: () => void;
+  refine: (query: string) => void;
+  voiceListeningState: VoiceListeningState;
+  templateContext: TemplateContext;
+};
+
+@Component({
+  selector: 'ais-voice-search',
+  template: `
+    <div [class]="cx()">
+      <button
+        type="button"
+        [class]="cx('button')"
+        [title]="state.isBrowserSupported ? buttonTitle : disabledButtonTitle"
+        [disabled]="!state.isBrowserSupported"
+        (click)="handleClick($event)"
+      >
+        <ng-container *ngTemplateOutlet="button ? button : defaultButton; context: state.templateContext"></ng-container>
+      </button>
+      <div [class]="cx('status')">
+        <ng-container *ngTemplateOutlet="status ? status : defaultStatus; context: state.templateContext"></ng-container>
+      </div>
+    </div>
+
+    <ng-template #defaultButton let-status="status" let-errorCode="errorCode" let-isListening="isListening">
+      <svg
+        xmlns='http://www.w3.org/2000/svg'
+        width='16'
+        height='16'
+        viewBox='0 0 24 24'
+        fill='none'
+        stroke='currentColor'
+        strokeWidth='2'
+        strokeLinecap='round'
+        strokeLinejoin='round'
+      >
+        <ng-container *ngIf="isNotAllowedError(); then errorSvgContent else normalSvgContent">
```

Comment:
I'm thinking maybe since `isNotAllowedError` is not exposed in templateContext, anyone providing their own `button` template will need to reimplement it. Do you think that can be a problem?

Reply:
`isNotAllowedError` is a method returning a value out of `status` and `errorCode` which both are inside `templateContext`. Even if user wants the exact same behaviour as the default template, it will be easy enough, I think.

Reply:
ok, I agree. Nevermind 😄